app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -86,49 +74,50 @@ object BitcoindRpcBackendUtil extends Logging {
     ()
   }
 
-  /** Helper method to sync the wallet until the bitcoind height */
-  private def doSync(
-      walletHeight: Int,
-      bitcoindHeight: Int,
+  /** Helper method to sync the wallet until the bitcoind height.
+    * This method returns a Sink that you can give block heights too and
+    * the sink will synchronize our bitcoin-s wallet against bitcoind
+    */
+  private def buildBitcoindSyncSink(
       bitcoind: BitcoindRpcClient,
-      wallet: Wallet)(implicit system: ActorSystem): Future[Wallet] = {
-    if (walletHeight > bitcoindHeight) {
-      val msg = s"Bitcoind and wallet are in incompatible states, " +
-        s"wallet height: $walletHeight, bitcoind height: $bitcoindHeight"
-      logger.error(msg)
-      Future.failed(new RuntimeException(msg))
-    } else {
-      logger.info(s"Syncing from $walletHeight to $bitcoindHeight")
-
-      import system.dispatcher
-
-      val hasFiltersF = bitcoind
-        .getFilter(wallet.walletConfig.chain.genesisHashBE)
-        .map(_ => true)
-        .recover { case _: Throwable => false }
-
-      val blockRange = walletHeight.to(bitcoindHeight).tail
-      val numParallelism = Runtime.getRuntime.availableProcessors()
-      logger.info(s"Syncing ${blockRange.size} blocks")
-      logger.info(s"Fetching block hashes")
-      val hashFs = Source(blockRange)
-        .mapAsync(numParallelism) {
-          bitcoind
-            .getBlockHash(_)
-            .map(_.flip)
-        }
-        .toMat(Sink.seq)(Keep.right)
-        .run()
+      wallet: Wallet)(implicit
+      system: ActorSystem): Future[Sink[Int, Future[Wallet]]] = {
```

Comment:
Same concept here, return the `Sink` rather than the materialized stream represented by `Future[Wallet]`